### ISSUE TITLE
Add format `asciidoc` lead to NotFound error

### ISSUE THREAD

**Issue by YuanchenZhu2020** [quarto-dev/quarto-cli#5454]
### Bug description

Sorry for my reckless response in #5395. I should first create a minimal example to reproduce the problem. I'm not sure if this is a bug😵‍💫, but thanks for your response.

## Quarto Version

```
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.3.340
      Path: D:\R\Quarto\bin
      CodePage: 936

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.8.10
      Path: D:/Program Files/Python38/python.exe
      Jupyter: 5.2.0
      Kernels: python3

[>] Checking Jupyter engine render....OK

[>] Checking R installation...........OK
      Version: 4.2.1
      Path: D:/R/R-4.2.1
      LibPaths:
        - D:/R/R-4.2.1/library
      knitr: 1.42
      rmarkdown: 2.21

[>] Checking Knitr engine render......OK
```

## Operating System and IDE

```
Windows 11
RStudio 2023.03.0 Build 386
```

## Steps to reproduce

1. Open RStudio and create a new Quarto Book project.
2. Open `_quarto.yml` and add `asciidoc: default` to the `format:`.
3. Render Book.

The stack trace is:

```
ERROR: NotFound: 系统找不到指定的文件。 (os error 2)

NotFound: 系统找不到指定的文件。 (os error 2)
    at Object.readTextFileSync (deno:runtime/js/40_read_file.js:42:16)
    at ServeRenderManager.fileRenderHash (file:///D:/R/Quarto/bin/quarto.js:103114:33)
    at file:///D:/R/Quarto/bin/quarto.js:103092:46
    at Array.forEach (<anonymous>)
    at ServeRenderManager.onRenderResult (file:///D:/R/Quarto/bin/quarto.js:103089:23)
    at serveProject (file:///D:/R/Quarto/bin/quarto.js:103200:19)
    at async Command.fn (file:///D:/R/Quarto/bin/quarto.js:103742:13)
    at async Command.execute (file:///D:/R/Quarto/bin/quarto.js:8437:13)
    at async quarto (file:///D:/R/Quarto/bin/quarto.js:127535:5)
    at async file:///D:/R/Quarto/bin/quarto.js:127553:9
```

And my `_quarto.yml` file is like:

```yaml
project:
  type: book

book:
  title: "example"
  author: "Norah Jones"
  date: "2023/5/8"
  chapters:
    - index.qmd
    - intro.qmd
    - summary.qmd
    - references.qmd

bibliography: references.bib

format:
  html:
    theme: cosmo
  asciidoc: default

editor: visual
```

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.
- [X] Please provide the output of `quarto check` so we know which version of quarto and its dependencies you're running.

**Comment by dragonstyle** [comment]
I'm sorry that you've run into this issue. I am not able to reproduce the issue locally despite trying a number of things. Unicode characters in paths are a potential source of problems, but in my testing this didn't cause the issue. Can you do anything to help further pinpoint the potential cause of the issue that would help me reproduce it?

**Comment by YuanchenZhu2020** [comment]
@dragonstyle The entire project is in a path with no Spaces or Unicode characters, the same with R and Quarto. Compared to the Quarto Book template, I only changed the `_quarto.yml` file as shown above.

I also tried:

1. Change env and locale to English using:
    ```
    Sys.setenv(LANG ="en")
    Sys.setlocale(category = "LC_ALL",locale = "English_United States.utf8")
    ```
2. Change my Windows language and locale to English and UTF8
3. Reinstall Quarto

But you still get the same error.

Here's the other information:

```r
> Sys.getenv()
ALLUSERSPROFILE                        C:\ProgramData
APPDATA                                C:\Users\xiaob\AppData\Roaming
CARGO_HOME                             D:\Rust\.cargo
CLASSPATH                              .;D:\Program Files\Java\jdk1.8.0_351\lib\dt.jar;D:\Program
                                       Files\Java\jdk1.8.0_351\lib\tools.jar;
CLICOLOR_FORCE                         1
CommonProgramFiles                     C:\Program Files\Common Files
CommonProgramFiles(x86)                C:\Program Files (x86)\Common Files
CommonProgramW6432                     C:\Program Files\Common Files
COMPUTERNAME                           XUVDESKTOP
ComSpec                                C:\WINDOWS\system32\cmd.exe
CUDA_PATH                              C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6
CUDA_PATH_V11_6                        C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6
CURL_CA_BUNDLE                         D:/R/R-4.2.1/etc/curl-ca-bundle.crt
DISPLAY                                :0
DriverData                             C:\Windows\System32\Drivers\DriverData
FPS_BROWSER_APP_PROFILE_STRING         Internet Explorer
FPS_BROWSER_USER_PROFILE_STRING        Default
GFORTRAN_STDERR_UNIT                   -1
GFORTRAN_STDOUT_UNIT                   -1
GIT_ASKPASS                            rpostback-askpass
HOME                                   C:/Users/xiaob/Documents
HOMEDRIVE                              C:
HOMEPATH                               \Users\xiaob
JAVA_HOME                              D:\Program Files\Java\jdk1.8.0_351
LOCALAPPDATA                           C:\Users\xiaob\AppData\Local
LOGONSERVER                            \\XUVDESKTOP
MPLENGINE                              tkAgg
MSYS2_ENV_CONV_EXCL                    R_ARCH
NUMBER_OF_PROCESSORS                   20
NVTOOLSEXT_PATH                        C:\Program Files\NVIDIA Corporation\NvToolsExt\
OneDrive                               C:\Users\xiaob\OneDrive
OneDriveConsumer                       C:\Users\xiaob\OneDrive
ORIGINAL_XDG_CURRENT_DESKTOP           undefined
OS                                     Windows_NT
PATH                                   D:\Program Files\Python38\Scripts;D:\Program
                                       Files\Python38;C:\rtools42\x86_64-w64-mingw32.static.posix\bin;C:\rtools42\usr\bin;D:\R\R-4.2.1\bin\x64;D:\R\Quarto\bin;C:\Program
                                       Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files\NVIDIA GPU
                                       Computing Toolkit\CUDA\v11.6\bin;C:\Program Files\NVIDIA GPU Computing
                                       Toolkit\CUDA\v11.6\libnvvp;D:\Program Files\Python38\Scripts\;D:\Program
                                       Files\Python38\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program
                                       Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA
                                       Corporation\NVIDIA
                                       NvDLISR;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;D:\Program
                                       Files\Bandizip\;D:\Program Files\Pandoc\;C:\Program Files\NVIDIA
                                       Corporation\Nsight Compute 2022.1.1\;D:\Program
                                       Files\Java\jdk1.8.0_351\bin;D:\Program
                                       Files\Java\jdk1.8.0_351\jre\bin;D:\Program
                                       Files\nodejs\;D:\R\R-4.2.1\bin;C:\Program Files\Git\cmd;D:\Program
                                       Files\mingw64\bin;C:\Program Files\dotnet\;D:\Program
                                       Files\Calibre2\;C:\Program Files\Chez Scheme 9.5.8\bin\a6nt;C:\Program
                                       Files\Chez Scheme 9.5.8\bin\ta6nt;D:\Program Files
                                       (x86)\calibre2\;D:\Rust\.cargo\bin;C:\Users\xiaob\AppData\Local\Microsoft\WindowsApps;C:\Users\xiaob\AppData\Roaming\npm;D:\Program
                                       Files (x86)\Microsoft VS Code\bin;D:\Program Files
                                       (x86)\Nmap;D:\R\TinyTex\bin\windows;D:\R\Quarto\bin;D:\Program
                                       Files\RStudio\resources\app\bin\postback
PATHEXT                                .COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY;.PYW
PROCESSOR_ARCHITECTURE                 AMD64
PROCESSOR_IDENTIFIER                   Intel64 Family 6 Model 154 Stepping 3, GenuineIntel
PROCESSOR_LEVEL                        6
PROCESSOR_REVISION                     9a03
ProgramData                            C:\ProgramData
ProgramFiles                           C:\Program Files
ProgramFiles(x86)                      C:\Program Files (x86)
ProgramW6432                           C:\Program Files
PSModulePath                           C:\Program
                                       Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
PUBLIC                                 C:\Users\Public
PYTHONIOENCODING                       utf-8
R_ARCH                                 /x64
R_CLI_HAS_HYPERLINK_IDE_HELP           true
R_CLI_HAS_HYPERLINK_IDE_RUN            true
R_CLI_HAS_HYPERLINK_IDE_VIGNETTE       true
R_COMPILED_BY                          gcc 10.3.0
R_DOC_DIR                              D:/R/R-4.2.1/doc
R_HOME                                 D:/R/R-4.2.1
R_INCLUDE_DIR                          D:/R/R-4.2.1/include
R_LIBS_SITE                            D:/R/R-4.2.1/site-library
R_LIBS_USER                            C:\Users\xiaob\AppData\Local/R/win-library/4.2
R_RTOOLS42_PATH                        C:\rtools42/x86_64-w64-mingw32.static.posix/bin;C:\rtools42/usr/bin
R_RUNTIME                              ucrt
R_SHARE_DIR                            D:/R/R-4.2.1/share
R_USER                                 C:/Users/xiaob/Documents
RETICULATE_PYTHON_FALLBACK             D:/Program Files/Python38/python.exe
RMARKDOWN_MATHJAX_PATH                 D:/Program Files/RStudio/resources/app/resources/mathjax-27
RS_LOCAL_PEER                          \\.\pipe\12295-rsession
RS_LOG_LEVEL                           WARN
RS_RPOSTBACK_PATH                      D:/Program Files/RStudio/resources/app/bin/rpostback.exe
RS_SHARED_SECRET                       35979db6-b68a-44f0-9a42-0f7bc93678ca
RSTUDIO                                1
RSTUDIO_CLI_HYPERLINKS                 true
RSTUDIO_CONSOLE_COLOR                  256
RSTUDIO_CONSOLE_WIDTH                  132
RSTUDIO_MSYS_SSH                       D:/Program Files/RStudio/resources/app/bin/msys-ssh-1000-18
RSTUDIO_PANDOC                         D:/Program Files/RStudio/resources/app/bin/quarto/bin/tools
RSTUDIO_PROGRAM_MODE                   desktop
RSTUDIO_SESSION_PID                    12560
RSTUDIO_SESSION_PORT                   12295
RSTUDIO_USER_IDENTITY                  xiaob
RSTUDIO_WINUTILS                       D:/Program Files/RStudio/resources/app/bin/winutils
RTOOLS42_HOME                          C:\rtools42
RUSTUP_HOME                            D:\Rust\.rustup
SESSIONNAME                            Console
SSH_ASKPASS                            rpostback-askpass
SystemDrive                            C:
SystemRoot                             C:\WINDOWS
TEMP                                   C:\Users\xiaob\AppData\Local\Temp
TERM                                   xterm-256color
TMP                                    C:\Users\xiaob\AppData\Local\Temp
USERDOMAIN                             XUVDESKTOP
USERDOMAIN_ROAMINGPROFILE              XUVDESKTOP
USERNAME                               xiaob
USERPROFILE                            C:\Users\xiaob
windir                                 C:\WINDOWS
ZES_ENABLE_SYSMAN                      1
> Sys.getlocale()
[1] "LC_COLLATE=Chinese (Simplified)_China.utf8;LC_CTYPE=Chinese (Simplified)_China.utf8;LC_MONETARY=Chinese (Simplified)_China.utf8;LC_NUMERIC=C;LC_TIME=Chinese (Simplified)_China.utf8"
> sessionInfo()
R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

Matrix products: default

locale:
[1] LC_COLLATE=Chinese (Simplified)_China.utf8  LC_CTYPE=Chinese (Simplified)_China.utf8   
[3] LC_MONETARY=Chinese (Simplified)_China.utf8 LC_NUMERIC=C                               
[5] LC_TIME=Chinese (Simplified)_China.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.2.1 tools_4.2.1    tinytex_0.45   xfun_0.39 
```

**Comment by mcanouil** [comment]
Changing the locales in R won't change them at your system level.
Quarto does not run inside R, it runs ont its own and calls R or Python for computations.

**Comment by YuanchenZhu2020** [comment]
@mcanouil I changed my Windows language and locale to English UTF8, but the only change was that the language of error became English.

**Comment by mcanouil** [comment]
Do you have the message in English?

**Comment by YuanchenZhu2020** [comment]
The error message is the same as before.

```
ERROR: NotFound: The system cannot find the file specified. (os error 2)

NotFound: The system cannot find the file specified. (os error 2)
    at Object.readTextFileSync (deno:runtime/js/40_read_file.js:42:16)
    at ServeRenderManager.fileRenderHash (file:///D:/R/Quarto/bin/quarto.js:103114:33)
    at file:///D:/R/Quarto/bin/quarto.js:103092:46
    at Array.forEach (<anonymous>)
    at ServeRenderManager.onRenderResult (file:///D:/R/Quarto/bin/quarto.js:103089:23)
    at serveProject (file:///D:/R/Quarto/bin/quarto.js:103200:19)
    at async Command.fn (file:///D:/R/Quarto/bin/quarto.js:103742:13)
    at async Command.execute (file:///D:/R/Quarto/bin/quarto.js:8437:13)
    at async quarto (file:///D:/R/Quarto/bin/quarto.js:127535:5)
    at async file:///D:/R/Quarto/bin/quarto.js:127553:9
```

**Comment by mcanouil** [comment]
Weird.

Would you be able to push your test project to a GitHub repository, in case we are missing something when trying to reproduce?

**Comment by YuanchenZhu2020** [comment]
OK, it is on <https://github.com/YuanchenZhu2020/quarto_test>.

**Comment by YuanchenZhu2020** [comment]
Thank you for your help ~ I will check my local environment again.
In the quarto CLI, everything went well, both for `render` and `preview`. The message in `quarto preview` is:

```
A new release of Deno is available: 1.28.2 → 1.33.2 Run `deno upgrade` to install it.
Preparing to preview
[1/4] index.qmd
[2/4] intro.qmd
[3/4] summary.qmd
[4/4] references.qmd

Watching files for changes
Browse at http://localhost:6016/
GET: /
```

**Comment by YuanchenZhu2020** [comment]
> Thank you for your help ~ I will check my local environment again. In the quarto CLI, everything went well, both for `render` and `preview`. The message in `quarto preview` is:
> 
> ```
> A new release of Deno is available: 1.28.2 → 1.33.2 Run `deno upgrade` to install it.
> Preparing to preview
> [1/4] index.qmd
> [2/4] intro.qmd
> [3/4] summary.qmd
> [4/4] references.qmd
> 
> Watching files for changes
> Browse at http://localhost:6016/
> GET: /
> ```

But when I use `quarto preview <path-of-project> --render all` in quarto CLI, I get the same error:

```
PS F:\quarto-test> quarto preview f:/quarto-test --render all
Rendering:
[1/4] index.qmd


processing file: index.qmd

output file: index.knit.md



processing file: index.qmd

output file: index.knit.md

[2/4] intro.qmd


processing file: intro.qmd

output file: intro.knit.md



processing file: intro.qmd

output file: intro.knit.md

[3/4] summary.qmd


processing file: summary.qmd

output file: summary.knit.md



processing file: summary.qmd

output file: summary.knit.md

[4/4] references.qmd

ERROR: NotFound: 系统找不到指定的文件。 (os error 2)

NotFound: 系统找不到指定的文件。 (os error 2)
    at Object.readTextFileSync (deno:runtime/js/40_read_file.js:42:16)
    at ServeRenderManager.fileRenderHash (file:///D:/R/Quarto/bin/quarto.js:103114:33)
    at file:///D:/R/Quarto/bin/quarto.js:103092:46
    at Array.forEach (<anonymous>)
    at ServeRenderManager.onRenderResult (file:///D:/R/Quarto/bin/quarto.js:103089:23)
    at serveProject (file:///D:/R/Quarto/bin/quarto.js:103200:19)
    at async Command.fn (file:///D:/R/Quarto/bin/quarto.js:103742:13)
    at async Command.execute (file:///D:/R/Quarto/bin/quarto.js:8437:13)
    at async quarto (file:///D:/R/Quarto/bin/quarto.js:127535:5)
    at async file:///D:/R/Quarto/bin/quarto.js:127553:9
```

**Comment by YuanchenZhu2020** [comment]
I installed the latest quarto (RHEL 7 Tarball)  and R-4.2.3 environment in my Fedora 37, and `quarto preview --render all` reports almost the same error. The quarto and R environment are installed for the first time in this system.

```
[zyc@pcf quarto-test]$ quarto preview ./ --render all
Rendering:
[1/4] index.qmd


processing file: index.qmd
                                                                                                            
output file: index.knit.md



processing file: index.qmd
                                                                                                            
output file: index.knit.md

[2/4] intro.qmd


processing file: intro.qmd
                                                                                                            
output file: intro.knit.md



processing file: intro.qmd
                                                                                                            
output file: intro.knit.md

[3/4] summary.qmd


processing file: summary.qmd
                                                                                                            
output file: summary.knit.md



processing file: summary.qmd
                                                                                                            
output file: summary.knit.md

[4/4] references.qmd

ERROR: NotFound: No such file or directory (os error 2)

NotFound: No such file or directory (os error 2)
    at Object.readTextFileSync (deno:runtime/js/40_read_file.js:42:16)
    at ServeRenderManager.fileRenderHash (file:///home/zyc/Softwares/quarto-1.3.340/bin/quarto.js:103114:33)
    at file:///home/zyc/Softwares/quarto-1.3.340/bin/quarto.js:103092:46
    at Array.forEach (<anonymous>)
    at ServeRenderManager.onRenderResult (file:///home/zyc/Softwares/quarto-1.3.340/bin/quarto.js:103089:23)
    at serveProject (file:///home/zyc/Softwares/quarto-1.3.340/bin/quarto.js:103200:19)
    at async Command.fn (file:///home/zyc/Softwares/quarto-1.3.340/bin/quarto.js:103742:13)
    at async Command.execute (file:///home/zyc/Softwares/quarto-1.3.340/bin/quarto.js:8437:13)
    at async quarto (file:///home/zyc/Softwares/quarto-1.3.340/bin/quarto.js:127535:5)
    at async file:///home/zyc/Softwares/quarto-1.3.340/bin/quarto.js:127553:9
```

The `quarto check` is:

```
[changfengqiu@pcf quarto-test]$ quarto check

[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.340
      Path: /home/changfengqiu/Softwares/quarto-1.3.340/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.3
      Path: /usr/bin/python3
      Jupyter: 5.3.0
      Kernels: python3

(\) Checking Jupyter engine render....0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.2.3
      Path: /usr/lib64/R
      LibPaths:
        - /home/changfengqiu/R/x86_64-redhat-linux-gnu-library/4.2
        - /usr/local/lib/R/library
        - /usr/lib64/R/library
        - /usr/share/R/library
      knitr: 1.42
      rmarkdown: 2.21

[✓] Checking Knitr engine render......OK
```

**Comment by mcanouil** [comment]
I can reproduce the error when calling `quarto preview --render all`.
Why would you preview to "all"? I wouldn't even expect it to work.

Use `quarto render --to all` to render to all formats.

**Comment by YuanchenZhu2020** [comment]
@mcanouil Earlier in RStudio, I just pressed the "Render Book" button in "Build" pane and got this error. I don't know what quarto commands RStudio executes behind the scenes.

I don't know `quarto preview --render all` until I try to use VSCode to reproduce this error yesterday. When I use `Quarto: Render Project` in VSCode Command Palette, I noticed  that `quarto preview --render all` was running in terminal, and I got this error again.
